sys/include/net/sock/dtls.h

```diff
@@ -0,0 +1,645 @@
+/*
+ * Copyright (C) 2019 HAW Hamburg
+ *                    Freie Universität Berlin
+ *                    Inria
+ *                    Daniele Lacamera
+ *                    Ken Bannister
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_sock_dtls    DTLS sock API
+ * @ingroup     net_sock net_dtls
+ * @brief       Sock submodule for DTLS
+ *
+ * DTLS sock acts as a wrapper for the underlying DTLS module to provide
+ * encryption for applications using the UDP sock API.
+ *
+ * How To Use
+ * ----------
+ *
+ * ### Summary
+ *
+ * - Include module implementing the DTLS sock API in the Makefile
+ * - Add credentials
+ *   1. Fill credman_credential_t with the credential information
+ *   2. Add the credential using @ref credman_add()
+ * - Server operation
+ *   1. Create UDP sock @ref sock_udp_create()
+ *   2. Create DTLS sock @ref sock_dtls_create()
+ *   3. Start listening with @ref sock_dtls_recv()
+ * - Client operation
+ *   1. Create UDP sock @ref sock_udp_create()
+ *   2. Create DTLS sock @ref sock_dtls_create()
+ *   3. Create session to server @ref sock_dtls_session_create()
+ *   4. Send packet to server @ref sock_dtls_send()
+ *
+ * ## Makefile Includes
+ *
+ * First, we need to [include](@ref including-modules) a module that implements
+ * this API in our applications Makefile. For example the module that
+ * implements this API for [tinydtls](@ref pkg_tinydtls) is called
+ * `tinydtls_sock_dtls'.
+ *
+ * The corresponding [pkg](@ref pkg) providing the DTLS implementation will be
+ * automatically included so there is no need to use `USEPKG` to add the pkg
+ * manually.
+ *
+ * Each DTLS implementation may have its own configuration options and caveat.
+ * This can be found at @ref net_dtls.
+ *
+ * ### Adding credentials
+ *
+ * Before using this API, either as a server or a client, we need to first
+ * add the credentials to be used for the encryption using
+ * [credman](@ref net_credman). Note that credman does not copy the credentials
+ * given into the system, it only has information about the credentials and
+ * where it is located at. So it is your responsibility to make sure that the
+ * credential is valid during the lifetime of your application.
+ *
+ * Instruction on how to generate the credentials can be found here [TODO].
```

Comment:
Maybe remove this sentence for now and re-add it once the doc on how to generate credentials is there.